Homework 3/Homeworks MS/Search/Search.c

```diff
@@ -0,0 +1,186 @@
+#include <stdio.h>
+#include <time.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+int* makeAnArray(int lengthOfArray)
+{
+    return calloc(lengthOfArray, sizeof(int));
+}
+ 
+void makeAnArrayOfRandomNumbers(int array[], int lengthOfArray)
+{
+    srand((unsigned)time(0));
+    for (int i = 0; i < lengthOfArray; ++i)
+    {
+        array[i] = -100 + rand() % 200;
+    }
+}
+
+int findTheSupportElement(int array[], int stopIndex, int startIndex)
+{
+    int i = startIndex + 1;
+    while (array[i - 1] == array[i] && i <= stopIndex)
+    {
+        ++i;
+    }
+    int supportElement = array[i] > array[i - 1] ? array[i] : array[i - 1];
```

Comment:
```suggestion
    const int supportElement = array[i] > array[i - 1] ? array[i] : array[i - 1];
```
А то и просто `return array[i] > array[i - 1] ? array[i] : array[i - 1];`